plugins/outputs/socket_writer/socket_writer.go

```diff
@@ -58,12 +70,26 @@ func (sw *SocketWriter) SetSerializer(s serializers.Serializer) {
 }
 
 func (sw *SocketWriter) Connect() error {
+	var (
+		c   net.Conn
+		err error
+	)
```

Comment:
nit: it's idiomatic go to declare variables right before they're used, not at the top of the function.
Also `err` doesn't need to be declared at all. It'll get declared on line 83.